File: engine/engine-model/src/main/scala/tools/packet/ui/GetKeyMap.scala
package tools.packet.ui

import client.KeyBinding
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GetKeyMap(private var _bindings: java.util.Map[java.lang.Integer, KeyBinding]) extends PacketInput {
  def bindings: java.util.Map[java.lang.Integer, KeyBinding] = _bindings

  override def opcode(): SendOpcode = SendOpcode.KEYMAP
}

File: engine/engine-model/src/main/java/tools/packet/ui/GetKeyMap.java
package tools.packet.ui;

import java.util.Map;

import client.KeyBinding;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GetKeyMap(Map<Integer, KeyBinding> bindings) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.KEYMAP;
   }
}